### ISSUE TITLE
Qwen3.5/3.8 LoRA target map skips Gated DeltaNet projections

### ISSUE THREAD

**Issue by ekzhu** [agentscope-ai/TuFT#149]
## Summary

On current `main`, TuFT's Qwen LoRA target map silently skips every linear projection in the Gated DeltaNet blocks used by Qwen3.5 and Qwen3.8.

Both model families use three linear-attention blocks for every full-attention block. With the default public modifiers (`train_attn=True`, `train_mlp=True`), TuFT resolves only:

```text
q_proj, k_proj, v_proj, o_proj, gate_proj, up_proj, down_proj
```

The full-attention and MLP names match, but the Gated DeltaNet blocks use:

```text
linear_attn.in_proj_qkv
linear_attn.in_proj_z
linear_attn.in_proj_b
linear_attn.in_proj_a
linear_attn.out_proj
```

None of those modules receive LoRA adapters.

This is separate from #141: that issue made HF and FSDP honor the same target geometry, while this issue concerns the Qwen geometry that both backends now resolve.

## Verified environment

- TuFT `main`: `146839b144ea4994428b4f8239b42e385fb88a69`
- Commit subject: `fix(fsdp): honor LoRA target modifiers (#144)`
- GPU: NVIDIA A100-SXM4-80GB
- Transformers: 5.5.4
- PEFT: 0.19.1
- Official configs:
  - `Qwen/Qwen3.5-4B`
  - `Qwen/Qwen3.8-27B`

The verification used a clean detached worktree created after fetching `origin/main`.

## Exact coverage from official model configs

| Model | Full-attention blocks | Linear-attention blocks | Full-attention linear modules matched | Gated DeltaNet linear modules matched | Gated DeltaNet linear weights matched |
|---|---:|---:|---:|---:|---:|
| Qwen3.5-4B | 8 | 24 | 32 / 32 | **0 / 120** | **0 / 1,010,565,120** |
| Qwen3.8-27B | 16 | 48 | 64 / 64 | **0 / 240** | **0 / 5,560,074,240** |

MLP coverage remains complete: 96/96 modules for 4B and 192/192 for 27B.

## A100 LoRA injection smoke test

I also instantiated architecture-faithful reduced-width `Qwen3_5ForConditionalGeneration` models using each official config's exact layer-type sequence, moved them to an A100, injected PEFT LoRA using TuFT's resolved target list, and ran a backward pass through a matched projection.

Results:

```text
Qwen3.5-4B layout:
  wrapped_full_attention=32
  wrapped_linear_attention=0
  wrapped_mlp=96
  wrapped_vision=0
  lora_B_grad_nonzero=true

Qwen3.8-27B layout:
  wrapped_full_attention=64
  wrapped_linear_attention=0
  wrapped_mlp=192
  wrapped_vision=0
  lora_B_grad_nonzero=true
```

This confirms that adapter injection and gradients work for the names TuFT selects, while no Gated DeltaNet projection is adapted.

## Minimal reproducer

```python
import torch
from accelerate import init_empty_weights
from transformers import AutoConfig
from transformers.models.qwen3_5.modeling_qwen3_5 import Qwen3_5ForCausalLM
from tuft.backends.lora_modules import resolve_target_modules

for model_id in ("Qwen/Qwen3.5-4B", "Qwen/Qwen3.8-27B"):
    cfg = AutoConfig.from_pretrained(model_id)
    targets = resolve_target_modules(
        model_id,
        train_attn=True,
        train_mlp=True,
        train_unembed=False,
    )

    with init_empty_weights():
        model = Qwen3_5ForCausalLM(cfg.text_config)

    linear_attn = [
        name
        for name, module in model.named_modules()
        if ".linear_attn." in name and isinstance(module, torch.nn.Linear)
    ]
    matched = [
        name
        for name in linear_attn
        if any(name == t or name.endswith("." + t) for t in targets)
    ]

    print(model_id, targets, len(linear_attn), len(matched))
```

Current output:

```text
Qwen/Qwen3.5-4B [...] 120 0
Qwen/Qwen3.8-27B [...] 240 0
```

## Impact

A Qwen3.5/3.8 training run can appear healthy while `train_attn=True` adapts attention projections in only one quarter of the text blocks. This changes LoRA capacity and makes "attention + MLP" results easy to misinterpret, especially for knowledge-acquisition and long-context experiments.

## Expected behavior

TuFT should either:

1. provide an architecture-aware Qwen3.5/3.8 target geometry that includes the five Gated DeltaNet `nn.Linear` projections; or
2. explicitly reject/warn about the incomplete geometry instead of silently treating it as full attention coverage.

A fix should avoid accidentally targeting the vision backbone and should keep HF training, FSDP slot allocation, checkpoint metadata, and vLLM export/load geometry aligned.

## Suggested acceptance criteria

- Qwen3.5/3.8 target resolution covers full-attention, MLP, and the five LoRA-compatible Gated DeltaNet linear projections when requested.
- The text target selection does not add adapters to the vision encoder.
- HF and FSDP resolve identical effective module sets.
- Saved adapter/checkpoint metadata records the expanded target set.
- Tests cover both the 32-layer Qwen3.5-4B pattern and 64-layer Qwen3.8-27B pattern.
- A save/reload and vLLM adapter smoke test passes with the expanded geometry.
